src/preferences/upgrade.cpp

```diff
@@ -33,78 +35,107 @@ Upgrade::~Upgrade() {
 
 namespace {
 // mapping to proactively move users to the new all-shader waveform types
-void upgradeToAllShaders(int* waveformType,
-        WaveformWidgetBackend* waveformBackend,
-        allshader::WaveformRendererSignalBase::Options* waveformOptions) {
-    if (*waveformBackend == WaveformWidgetBackend::AllShader) {
-        return;
-    }
-    *waveformOptions = allshader::WaveformRendererSignalBase::None;
-    *waveformBackend = WaveformWidgetBackend::AllShader;
+std::tuple<WaveformWidgetType::Type,
+        WaveformWidgetBackend,
+        allshader::WaveformRendererSignalBase::Options>
+upgradeToAllShaders(int unsafeWaveformType,
+        int unsafeWaveformBackend,
+        int unsafeWaveformOption) {
     // TODO: convert `WaveformWidgetType::Type` to an enum class then shorten more `using enum ...`
     using WWT = WaveformWidgetType;
 
+    if (static_cast<WaveformWidgetBackend>(unsafeWaveformBackend) ==
+            WaveformWidgetBackend::AllShader) {
+        allshader::WaveformRendererSignalBase::Options waveformOption =
+                static_cast<allshader::WaveformRendererSignalBase::Options>(
+                        unsafeWaveformOption) &
+                allshader::WaveformRendererSignalBase::All;
```

Comment:
I assume is `allshader::WaveformRendererSignalBase::All` means all options, right? This should be reflected in the naming (use `enum class`) in a future PR.

Reply:
It's new code so I'm fixing it now

Reply:
Updated now - is `AllOptionsCombined` good enough?

Reply:
Yeah.